ir/CMakeLists.txt

```diff
@@ -14,6 +14,7 @@
 
 set (IR_SRCS
   base.cpp
+  bitrange.cpp
```

Comment:
This seems like its own contribution? Maybe we can do it in a separate PR?

Reply:
Also it looks like there is no IR usage at all, can we make part of `lib`?

Reply:
Sure, I can make it a separate PR. It'll need to go in first since the def-use pass uses `le_bitrange`.

I initially made it part of lib but then switched to ir. There are a pair of functions for JSON serialization/deserialization that require the JSONGenerator/JSONLoader classes from ir, but this causes compilation failure in clean builds because lib is built before the ir generator runs. (The JSONGenerator/JSONLoader source files pull in ir.h.)

The Tofino backend IR uses le_bitrange/nw_bitrange in a handful of places, so it requires the ability to serialize to JSON. Here are a few ways I could deal with this:
* Strip JSON serialization code from the open-source bitrange code, and subclass in Tofino to add JSON serialization back in. This would require work to ensure the Tofino code always uses the subclass.
* Only two functions in bitrange.cpp use JSONGenerator/JSONLoader. I could place these two functions in ir/bitrange.cpp and have the rest of the code in lib. But this means that using p4ctoolkit library would require also linking in the ir library.
* Leave all the code in ir, even though the open source IR doesn't use the new bitrange classes.
Any better suggestions that these options?

Reply:
Done: #4496